forem/forem
# Avoid layout funkiness when non-breaking characters are used

## What type of PR is this? (check all applicable)


## Description

We had a recent bug on the home page where the use of some special characters in a post title was causing the layout to go very wide and wonky. This affected Safari especially, and a bit of investigation showed one of the root causes was the use of the CSS `overflow-wrap: anywhere;`, which is not supported in Safari. We added a fallback of `overflow-wrap: break-word;` (supported by Safari) as part of that fix.

This PR identifies other user-entered data that would cause layout problems if a user used these non-breaking spaces. It applies the same fallback mentioned above everywhere that we currently use `overflow-wrap: anywhere;`. Some context from [the docs](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap) regarding the `break-word` value:

Another way we fixed the layout in the bugs noted below was to make use of the `crayons-layout__content` class, which we'd already created for this precise reason of preventing grid column overlap. This PR fixes some other areas in the same way. 

## Related Tickets & Documents



Related to:

https://github.com/forem/forem/pull/16771
https://github.com/forem/forem/pull/16773

Fixes https://github.com/forem/forem/issues/16772

## QA Instructions, Screenshots, Recordings

NB: It seems like Safari is a bit more stringent about caching CSS styles... My advice would be to run the app in a private browsing window to make sure you're definitely looking at the new CSS.

Update the following things to include long text with non-breaking characters, like [those found in the title of this post](https://dev.to/inhuofficial/i-am-warning-you-dont-click-42im):

- A "suggested user" in the Onboarding's profile bio (update the user's bio, then go to Admin > Customization > Onboarding and add their username to "suggested users")
- The bio for any user who appears as an author in the home feed (and would therefore have a profile preview card)
- The title and description of a Listing
- The name of the Forem (set in Admin > Customization > Community content)
- The name of an organization
- The title of a published post


Visit the following areas in Safari:

- The onboarding flow "suggested users" page (you should see the user whose bio you edited above)
- The home feed preview card for the user whose bio you updated
- The Forem name in the top left corner
- The homepage left sidebar when logged out
- The "sign up" break in the home feed when logged out
- The listings sidebar on the home page
- The listings page
- The user profile of someone who's a member of the organization you edited
- The new admin member detail view for someone who's a member of the organization you edited
- A comment permalink (i.e. the individual permalink for a comment) from the post with the funky title you created

Have a click around the app generally after adding all those horrible non-breaking characters just to triple check I've not missed something!


### UI accessibility concerns?

N/A

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
